docs/latest/design/extending-envoy-gateway.md

```diff
@@ -144,6 +144,62 @@ latency between Envoy and the authentication service and also allows the end use
 
 ![Control Plane Extension Example](../images/extension-auth-example.png)
 
+## Design Decisions
+
+- Since an HTTPRoute resource corresponds to an [Envoy Proxy Route resource][], the name of the Route is set using the name and namespace of the HTTPRoute that generated it so that any extension looking at the
+xDS resources provided by Envoy Gateway can have the same context about which Routes to modify.
+- The responsibility of managing scale and performance of hook calls and modification of xDS resources falls on the extension controller, not Envoy Gateway
+
+## Example: Post xDS Hook API
+
+```go
+// TODO: If Envoy Gateway is modified to watch resources introduced by extensions, then this API would be updated to send those resources over to the extension
+// for any hook calls.
+type PostHTTPListenerTranslationRequest struct {
+    state         protoimpl.MessageState
+    sizeCache     protoimpl.SizeCache
+    unknownFields protoimpl.UnknownFields
+
+    // HTTPListenerExtensionContext provides context on
+    // the user's intent so that the provided XDS can be modified by the Extension.
+    // It includes sub-fields that provide context about an HTTPRoute containing an
+    // ExtensionRef HTTPRouteFilterType. The context provided includes which filterchain/s,
+    // route and Group/Kind of the object referent.
+    ExtensionContext *HTTPListenerExtensionContext `protobuf:"bytes,1,opt,name=extension_context,json=extensionContext,proto3" json:"extension_context,omitempty"`
+    Listener         *v3.Listener                  `protobuf:"bytes,2,opt,name=listener,proto3" json:"listener,omitempty"`
+    RouteTable       *v31.RouteConfiguration       `protobuf:"bytes,3,opt,name=route_table,json=routeTable,proto3" json:"route_table,omitempty"`
+}
+
+
+type PostHTTPListenerTranslationResponse struct {
+    state         protoimpl.MessageState
+    sizeCache     protoimpl.SizeCache
+    unknownFields protoimpl.UnknownFields
+
+    Clusters   []*v32.Cluster          `protobuf:"bytes,1,rep,name=clusters,proto3" json:"clusters,omitempty"`
+    Listener   *v3.Listener            `protobuf:"bytes,2,opt,name=listener,proto3" json:"listener,omitempty"`
+    RouteTable *v31.RouteConfiguration `protobuf:"bytes,3,opt,name=route_table,json=routeTable,proto3" json:"route_table,omitempty"`
+    Secrets    []*v33.Secret           `protobuf:"bytes,4,rep,name=secrets,proto3" json:"secrets,omitempty"`
+}
+
+service EnvoyGatewayExtension {
```

Comment:
it's a bit confusing to follow since you are mixing the protobuf specification & the generated go structs and including both is sort of redundant information. I would just chose one or the other. I would personally go with just the proto specification since it contains all the necessary design info + API specification and so should be more straightforward to follow

Reply:
I agree with this. I would go with the Protobuf specification because the other stuff is just noise.